2025-06-11 07:49:45 +0000 - dagster - DEBUG - __ASSET_JOB - 1007b48d-e811-460c-8ca7-6c0ad7db2e85 - 2347 - LOGS_CAPTURED - Started capturing logs in process (pid: 2347).
2025-06-11 07:49:45 +0000 - dagster - DEBUG - __ASSET_JOB - 1007b48d-e811-460c-8ca7-6c0ad7db2e85 - 2347 - get_monthly_lotto_result - STEP_START - Started execution of step "get_monthly_lotto_result".
2025-06-11 07:49:45,771 - INFO - Context impl SQLiteImpl.
2025-06-11 07:49:45,772 - INFO - Will assume non-transactional DDL.
2025-06-11 07:50:17 +0000 - dagster - ERROR - __ASSET_JOB - 1007b48d-e811-460c-8ca7-6c0ad7db2e85 - 2347 - get_monthly_lotto_result - STEP_FAILURE - Execution of step "get_monthly_lotto_result" failed.

dagster._core.errors.DagsterExecutionStepExecutionError: Error occurred while executing op "get_monthly_lotto_result"::

playwright._impl._errors.TimeoutError: ElementHandle.select_option: Timeout 30000ms exceeded.
Call log:
  - attempting select option action
    2 × waiting for element to be visible and enabled
      - did not find some options
    - retrying select option action
    - waiting 20ms
    2 × waiting for element to be visible and enabled
      - did not find some options
    - retrying select option action
      - waiting 100ms
    58 × waiting for element to be visible and enabled
       - did not find some options
     - retrying select option action
       - waiting 500ms


Stack Trace:
  File "/app/.venv/lib/python3.12/site-packages/dagster/_core/execution/plan/utils.py", line 56, in op_execution_error_boundary
    yield
  File "/app/.venv/lib/python3.12/site-packages/dagster/_utils/__init__.py", line 392, in iterate_with_context
    next_output = next(iterator)
                  ^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 129, in _coerce_op_compute_fn_to_iterator
    result = invoke_compute_fn(
             ^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 117, in invoke_compute_fn
    return fn(context, **args_to_pass) if context_arg_provided else fn(**args_to_pass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/src/lotto_project/assets/lotto_result.py", line 101, in get_monthly_lotto_result
    page.wait_for_selector("id=cphContainer_cpContent_ddlStartDate").select_option(value=day_from)
  File "/app/.venv/lib/python3.12/site-packages/playwright/sync_api/_generated.py", line 2291, in select_option
    self._sync(
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_sync_base.py", line 115, in _sync
    return task.result()
           ^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_element_handle.py", line 164, in select_option
    return await self._channel.send("selectOption", params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_connection.py", line 61, in send
    return await self._connection.wrap_api_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_connection.py", line 528, in wrap_api_call
    raise rewrite_error(error, f"{parsed_st['apiName']}: {error}") from None

